internal/services/network/network_manager_commit_resource.go

```diff
@@ -0,0 +1,348 @@
+			networkManagerId, err := parse.NetworkManagerID(state.NetworkManagerId)
+			if err != nil {
+				return err
+			}
+
+			normalizedLocation := azure.NormalizeLocation(state.Location)
+			id := parse.NewNetworkManagerCommitID(networkManagerId.SubscriptionId, networkManagerId.ResourceGroup, networkManagerId.Name, normalizedLocation, state.ScopeAccess)
+
+			metadata.Logger.Infof("creating %s", id)
+
+			listParam := network.ManagerDeploymentStatusParameter{
+				Regions:         &[]string{normalizedLocation},
+				DeploymentTypes: &[]network.ConfigurationType{network.ConfigurationType(state.ScopeAccess)},
+			}
+			resp, err := statusClient.List(ctx, listParam, id.ResourceGroup, id.NetworkManagerName, nil)
+			if err != nil {
+				if utils.ResponseWasNotFound(resp.Response) {
+					return metadata.ResourceRequiresImport(r.ResourceType(), id)
+				}
+				return fmt.Errorf("checking for the presence of an existing %s: %+v", id, err)
+			}
+			if resp.Value != nil && len(*resp.Value) != 0 && *(*resp.Value)[0].ConfigurationIds != nil && len(*(*resp.Value)[0].ConfigurationIds) != 0 {
+				return metadata.ResourceRequiresImport(r.ResourceType(), id)
+			}
+
+			input := network.ManagerCommit{
+				ConfigurationIds: &state.ConfigurationIds,
+				TargetLocations:  &[]string{state.Location},
+				CommitType:       network.ConfigurationType(state.ScopeAccess),
+			}
+
+			if _, err := client.Post(ctx, input, id.ResourceGroup, id.NetworkManagerName); err != nil {
+				return fmt.Errorf("creating %s: %+v", id, err)
+			}
+
+			if err = resourceManagerCommitWaitForFinished(ctx, statusClient, id, metadata.ResourceData); err != nil {
+				return err
+			}
+
+			metadata.SetID(id)
+			return nil
+		},
+		Timeout: 2 * time.Hour,
+	}
+}
+
+func (r ManagerCommitResource) Read() sdk.ResourceFunc {
+	return sdk.ResourceFunc{
+		Func: func(ctx context.Context, metadata sdk.ResourceMetaData) error {
+			client := metadata.Client.Network.ManagerDeploymentStatusClient
+			id, err := parse.NetworkManagerCommitID(metadata.ResourceData.Id())
+			if err != nil {
+				return err
+			}
+
+			metadata.Logger.Infof("retrieving %s", *id)
+
+			listParam := network.ManagerDeploymentStatusParameter{
+				Regions:         &[]string{id.Location},
+				DeploymentTypes: &[]network.ConfigurationType{network.ConfigurationType(id.ScopeAccess)},
+			}
+			resp, err := client.List(ctx, listParam, id.ResourceGroup, id.NetworkManagerName, nil)
+			if err != nil {
+				if utils.ResponseWasNotFound(resp.Response) {
+					metadata.Logger.Infof("%s was not found - removing from state!", *id)
+					return metadata.MarkAsGone(id)
+				}
+				return fmt.Errorf("retrieving %s: %+v", *id, err)
+			}
+
+			if resp.Value == nil || len(*resp.Value) == 0 || *(*resp.Value)[0].ConfigurationIds == nil || len(*(*resp.Value)[0].ConfigurationIds) == 0 {
+				metadata.Logger.Infof("%s was not found - removing from state!", *id)
+				return metadata.MarkAsGone(id)
+			}
+
+			commit := (*resp.Value)[0]
```

Comment:
This is checked in above line at https://github.com/hashicorp/terraform-provider-azurerm/pull/20451/files#diff-f7faa7358d61518bec904e6d50a8453bf5220356bdae2d40b01027c61babfabeR170

Reply:
The preceding check only checks for a non-nil pointer to a greater than zero length slice - I meant we should check that the slice elements are exactly 1 and not higher, since this would be unexpected and increase the risk of something going wrong.

Reply:
thanks, this makes sense.